spec/unit/verifier_spec.rb

```diff
@@ -75,6 +75,44 @@
         }.to output(expected_output).to_stdout
       end
     end
+
+    context "one warning" do
+      before do
+        subject.name = "xyz"
+        subject.warnings.push("a warning")
+      end
+
```

Comment:
Maybe you should have added a test here to test the `has_warnings?` method. This would have prevented the wrong logic. Something like:

```ruby
it "has warning" do
  expect(subject.has_warning?).to be(true)
end
```